### PR TITLE
[GNA] Support of the tensor names for scale factors

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
Starting from IRv11 get_any_names() returns the tensor name instead of the friendly node name and to support that I added searching of the scale factors by any tensor name

### Tickets:
 - *98758*

### CODE REVIEW

**Review by nkogteva — APPROVED** [review]

**Review by kbruniec — COMMENTED** [review]
Make sense,
However, gna_plugin.cpp is quite large, would move this logic to some less exposed file... 
, eg https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gna/src/scale_factor_helper.cpp looks fine for this.
After moving to some other file I would mark the helper function as deprecated (or similar) and add TODO (eg saying to be removed after fully switching to POT optimized models).

Beside, It would be good to have (unit) test covering this change.

**Review by kbruniec — APPROVED** [review]
LGTM

File: src/plugins/intel_gna/src/scale_factor_helper.hpp [hunk 1]
```diff
@@ -19,9 +19,8 @@ class Config;
  */
 namespace helpers {
 
-void ApplyInputScaleFactors(const Config& config,
-                            const header_latest::ModelHeader& header,
-                            GnaInputs& inputs);
+void ApplyInputScaleFactors(GnaInputs& inputs, const Config& config, const header_latest::ModelHeader& header);
+void ApplyInputScaleFactors(GnaInputs& inputs, const Config& config);
```
Comment (marcinkusm) [inline src/plugins/intel_gna/src/scale_factor_helper.hpp:23]: just wonder if we need to expose this method in header if it should not be used anywhere beside internally in source file by this method
```
void ApplyInputScaleFactors(GnaInputs& inputs, const Config& config, const header_latest::ModelHeader& header);
```

File: src/plugins/intel_gna/src/scale_factor_helper.cpp [hunk 2]
```diff
@@ -98,11 +110,7 @@ void ApplyInputScaleFactors(const Config& config,
         return;
     }
 
-    if (!CheckIfCanApplyCustomScaleFactor(header)) {
-        return;
-    }
```
[the remark this answers is not in the document]
Comment (marcinkusm) [reply src/plugins/intel_gna/src/scale_factor_helper.cpp:103]: Just afraid that if there will be imported older version of model ( <= 2.7 )  the warning will be visible even if there will be no custom scale factors:

```
 if (!header_latest::IsFirstVersionLower(header.version, sc_forbid_override_scale_factor)) {
        ::log::warning() << "Cannot apply custom scale factor for model versions >= "
<< sc_forbid_override_scale_factor.major << "." << sc_forbid_override_scale_factor.minor
<< std::endl;
```

I suggest to leave it as it was or eventually change message: e.g.
The for this version of model only built in scale factors supported.

[the remark this answers is not in the document]
Comment (mryzhov) [reply src/plugins/intel_gna/src/scale_factor_helper.cpp]: I've updated the code to support this case, it causes some code duplication

**Review by marcinkusm — APPROVED** [review]

**Pull request merged**